src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+        {
+            i |= getBoundingBoxIndex(EnumFacing.NORTH);
+        }
+
+        if (state.getValue(EAST))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.EAST);
+        }
+
+        if (state.getValue(SOUTH))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.SOUTH);
+        }
+
+        if (state.getValue(WEST))
+        {
+            i |= getBoundingBoxIndex(EnumFacing.WEST);
+        }
+
+        return i;
+    }
+
+    /**
+     * Get the actual Block state of this Block at the given position. This applies properties not visible in the
+     * metadata, such as fence connections.
+     */
+    @NotNull
+    public IBlockState getActualState(@NotNull final IBlockState Tempstate,
+                                      @NotNull final IBlockAccess worldIn,
+                                      @NotNull final BlockPos pos)
+    {
+        return Tempstate.withProperty(NORTH, canPaneConnectTo(worldIn, pos, EnumFacing.NORTH))
+                .withProperty(SOUTH, canPaneConnectTo(worldIn, pos, EnumFacing.SOUTH))
+                .withProperty(WEST, canPaneConnectTo(worldIn, pos, EnumFacing.WEST))
+                .withProperty(EAST, canPaneConnectTo(worldIn, pos, EnumFacing.EAST));
+    }
+
+    /**
+    * Used to determine ambient occlusion and culling when rebuilding chunks for render
+    */
+
+    public boolean isOpaqueCube(@NotNull final IBlockState state)
+    {
+        return false;
+    }
+
+    public boolean isFullCube(@NotNull final IBlockState state)
+    {
+        return false;
+    }
+
+    private boolean canPaneConnectToBlock(final Block blockIn)
+    {
+        return blockIn.getDefaultState().isFullCube()
+                || blockIn == this || blockIn == Blocks.GLASS || blockIn == Blocks.STAINED_GLASS ||
+                blockIn == Blocks.STAINED_GLASS_PANE || blockIn instanceof BlockPaperwall;
+    }
+
+    @SideOnly(Side.CLIENT)
+    public boolean shouldSideBeRendered(@NotNull final IBlockState blockState,
+                                        @NotNull final IBlockAccess blockAccess,
+                                        @NotNull final BlockPos pos,
+                                        @NotNull final EnumFacing side)
+    {
+        return blockAccess.getBlockState
+                (pos.offset(side)).getBlock() != this && super.shouldSideBeRendered(blockState, blockAccess, pos, side);
+    }
+
+    @NotNull
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.TRANSLUCENT;
+    }
+
+    /**
+     * Convert the BlockState into the correct metadata value
+     */
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return 0;
+    }
+
+    /**
+     * Returns the blockstate with the given rotation from the passed blockstate. If inapplicable, returns the passed
+     * blockstate.
+     */
+    @NotNull
+    public IBlockState withRotation(@NotNull final IBlockState Tempstate,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)